### PR TITLE
Fix splinterd so it correctly loads the TOML configuration file from `SPLINTER_HOME`

### PR DESCRIPTION
**Pull request by leebradley**
Previously, splinterd would not take into consideration the value of
SPLINTER_HOME or SPLINTER_CONFIG_DIR with respect to the configuration
file. This change makes it so that splinterd will check both of those
values before loading the configuration.

### CODE REVIEW

**Review by Caleb-Hill — DISMISSED** [review]

**Comment by peterschwarz** [comment]
Tested, works as expected

File: splinterd/src/config/env.rs [hunk 1]
```diff
@@ -44,23 +44,69 @@ const METRICS_USERNAME_ENV: &str = "SPLINTER_INFLUX_USERNAME";
...
+/// Trait that outlines a basic read-only environment variable store
+pub trait EnvStore {
+    /// Returns an environment variable for the given key. Results are intended to mirror the
+    /// functionality of `env::var(key)`.
+    ///
+    /// # Arguments
+    ///
+    /// * `key` - A string slice that holds the name of the environment variable
+    ///
+    fn get(&self, key: &str) -> Result<String, VarError>;
```
Comment (peterschwarz) [inline splinterd/src/config/env.rs:56]: Given we never use the error, should this just return `Option<String>`?

**Review by peterschwarz — COMMENTED** [review]
Can this be split into two PRs? Both changes have value, but can be reviewed in isolation (the bug fix will likely be merged more quickly than the test change one).

File: splinterd/src/config/env.rs (hunk 1 above)
Comment (leebradley) [reply splinterd/src/config/env.rs]: Updated!

**Review by vaporos — CHANGES_REQUESTED** [review]

File: splinterd/src/main.rs [hunk 2]
```diff
@@ -712,10 +712,25 @@ fn setup_metrics_recorder(config: &Config) -> Result<(), UserError> {
...
-    matches
-        .value_of("config")
-        .unwrap_or("/etc/splinter/splinterd.toml")
+fn get_config_file(matches: &'_ ArgMatches) -> Result<String, UserError> {
+    if let Some(value) = matches.value_of("config") {
+        return Ok(value.to_string());
+    }
+
+    if let Ok(value) = env::var("SPLINTER_CONFIG_DIR") {
+        return Ok(value);
```
[the remark this answers is not in the document]
Comment (leebradley) [reply splinterd/src/main.rs]: Fixed!

File: splinterd/src/main.rs [hunk 3]
```diff
@@ -712,10 +712,25 @@ fn setup_metrics_recorder(config: &Config) -> Result<(), UserError> {
...
+    }
+
+    if let Ok(value) = env::var("SPLINTER_HOME") {
+        return match Path::new(&value).join("conf").to_str() {
```
[the remark this answers is not in the document]
Comment (leebradley) [reply splinterd/src/main.rs]: Fixed!

File: splinterd/src/main.rs [hunk 4]
```diff
@@ -712,10 +712,25 @@ fn setup_metrics_recorder(config: &Config) -> Result<(), UserError> {
...
+            Some(value) => Ok(value.to_string()),
+            None => Err(UserError::InvalidArgument(
+                "Could parse SPLINTER_HOME path".to_string(),
```
[the remark this answers is not in the document]
Comment (leebradley) [reply splinterd/src/main.rs]: Updated to better explain the error

**Comment by leebradley** [comment]
> Can this be split into two PRs? Both changes have value, but can be reviewed in isolation (the bug fix will likely be merged more quickly than the test change one).

Split off environment config testing changes into https://github.com/Cargill/splinter/pull/1525

**Review by vaporos — APPROVED** [review]

**Pull request merged**